src/algebra/order/floor.lean

```diff
@@ -839,6 +843,10 @@ lemma ceil_eq_iff : ⌈a⌉ = z ↔ ↑z - 1 < a ∧ a ≤ z :=
 by rw [←ceil_le, ←int.cast_one, ←int.cast_sub, ←lt_ceil, int.sub_one_lt_iff, le_antisymm_iff,
   and.comm]
 
+@[simp]
+lemma ceil_eq_self (h : ∃ z : ℤ, (z : α) = a) : (⌈a⌉ : α) = a :=
```

Comment:
This seems like a strange lemma to me, especially as a simp lemma. Perhaps instead

```suggestion
lemma ceil_int_cast (z : ℤ) : ⌈(z : α)⌉  = z :=
```
But I expect we already have that?

Reply:
That sounds good to me. It's pretty unusual to state theorem statements with `exists` in hypotheses; and then once you split it into two hypotheses, you end up with an equality, which again is unusual to have in a theorem statement.